### PR TITLE
loop conversion #2

### PR DESCRIPTION
**Pull request by ajax16384**
use auto loop
replace "erase(iter++)" with "iter = erase(iter)"

### CODE REVIEW

File: src/file_pool.cpp [hunk 1]
```diff
@@ -239,16 +239,18 @@ namespace libtorrent
 	{
 		std::unique_lock<std::mutex> l(m_mutex);
 
-		auto begin = m_files.lower_bound(std::make_pair(st, file_index_t(0)));
+		auto const begin = m_files.lower_bound(std::make_pair(st, file_index_t(0)));
+		auto current = begin;
```
Comment (arvidn) [inline src/file_pool.cpp]: if you're introducing a loop variable, I think the loop below should be turned into a for-loop. (and ``current`` could be named ``i``)

**Comment by arvidn** [comment]
lgtm apart from comment

File: src/file_pool.cpp (hunk 1 above)
Comment (ajax16384) [inline src/file_pool.cpp]: ok

**Comment by arvidn** [comment]
lgtm

**Comment by arvidn** [comment]
https://travis-ci.org/arvidn/libtorrent/jobs/210113639#L2096

**Comment by arvidn** [comment]
lgtm

**Pull request merged**